app/services/course/assessment/submission/zip_download_service.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+class Course::Assessment::Submission::ZipDownloadService
```

Comment:
Not sure if this service can be restructured to fit the following use cases: 
 1. Staff downloads all submissions from 1 assessment
 1. Staff downloads some submissions from 1 assessment
 1. Student downloads own submission from 1 assessment
 1. Student downloads all of their submissions from all assessments.

I think the last case can be handled in a separate service though, but if common parts can be factored out, we can do that. 👍